AUTOCALIBRATE command for Sniff-ISO15693 application

cacke-r opened the pull request:
This adds code to the Sniff15693 Application to run the AUTOCALIBRATE command.
But default, the Sniff15693 codec uses an Autothreshold detection - if this works fine, it shall be used.
If not you can disable it by un-commenting this line in the Makefile:
     #CONFIG_SETTINGS += -DSNIFF15693_DISABLE_AUTO_THRESHOLD

The AUTOCALIBRATION algorithm itself is explained in the source code (Application/Sniff15693.c)

cacke-r commented:
@ceres-c @fptrs pls find the PR here.
Looking forward for your review comments.
(I*ll be on vacation for few days - so i'll work on the findings only b/o next week)

fptrs commented on Firmware/Chameleon-Mini/Application/Sniff15693.c:
```diff
@@ -27,11 +100,146 @@ void SniffISO15693AppTask(void)
+                        }
+                        last_cycle_successful = false;
+                    }else{
+                        /* We have received card data now */
```
I think we should check the CRC first before assume that we received valid data.

cacke-r replied:
Good point. I though about this - and then dropped it. 
Will add the Crc check.

cacke-r replied:
Fixed with latest commit:
[Sniff15693: Add CRC Check on received data]

cacke-r commented:
@ceres-c @fptrs 
The PR was updated automatically. I've added the command - but I'm not yet done.
I need to rework the other patches as well. Will do this tomorrow and send you a ping here, once I'm done.
If you want you can already review the 2 new patches - they are ready. It is just the rest which needs rework now.

ceres-c commented:
Hi, sorry for the delay,
I like this approach, it now sounds user-friendly enough to me :)

cacke-r commented:
@ceres-c @fptrs 
so, finally I've added a command to control the autothreshold feature of the codec and removed the ifdefs.
I needed to re-arrange the patches a bit. 
Looking forward for further feedback ... or a merge :-)

cacke-r commented:
@ceres-c @fptrs 
Hey guys, I don't want to appear pushy 😀 but could you let me know I which timeframe I could expect further feedback ?

fptrs commented:
Hi @cacke-r,
sorry for the late reply. I'm going to test/review it this week. If everything is fine I'm going to merge it.

fptrs commented:
So I tested the autocalibration and it works fine. Very good job :+1: There is only one final request before I'm going to merge it. Can you change the autothreshold command to accept ```1``` and ```0``` as parameters instead of strings, similar to the ```field``` command.

cacke-r commented:
> So I tested the autocalibration and it works fine. Very good job +1 There is only one final request before I'm going to merge it. Can you change the autothreshold command to accept `1` and `0` as parameters instead of strings, similar to the `field` command.

@fptrs 
thanks for testing and reviewing this. I've just pushed the changes about ENABLE/DISABLE vs 0,1.

Yesterday I realized that there is still one issue with this autocalibrate. If we never receive
a valid frame - it will set the threshold to (0+0xFFF/2) - which is not so nice.
I'll work to fix it - means detect this as an error case - and set the threshold to the value it was before.

I'd like to raise this as a new PR - just to not open up this one again.
But if you prefer to have it as part of this PR - pls let me know.

fptrs commented:
I think the best solution is to include it here directly

cacke-r commented:
> I think the best solution is to include it here directly


OK, thanks for the feedback. Included the fix as well.
So - I think I'm done now :-)
